### PR TITLE
feat: update widget `state` when changing or updating dataTable

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [x] Not that difficult
- [ ] Approved feature branch merge to master

### CODE REVIEW

File: apps/web/src/common/modules/widgets/_components/WidgetFormOverlay.vue [hunk 1]
```diff
@@ -54,15 +47,13 @@ const state = reactive({
...
-        await fetcher({
-            widget_id: widgetGenerateState.widgetId,
-            data_table_id: widgetGenerateState.selectedDataTableId,
-        });
+        if (widgetGenerateState.widget?.data_table_id !== widgetGenerateState.selectedDataTableId) {
+            await widgetGenerateStore.updateWidget({
+                widget_id: widgetGenerateState.widgetId,
+                data_table_id: widgetGenerateState.selectedDataTableId,
+            });
+        }
```
Comment (piggggggggy) [inline apps/web/src/common/modules/widgets/_components/WidgetFormOverlay.vue:56]: Maybe, this need `// state: 'INACTIVE',` too

File: apps/web/src/common/modules/widgets/_components/WidgetFormOverlayStep2WidgetForm.vue [hunk 2]
```diff
@@ -184,15 +158,6 @@ const handleUpdateFieldValidation = (fieldName: string, isValid: boolean) => {
...
 
-/* Watcher */
-watch(() => widgetGenerateState.widgetValidMap, () => {
-    if (state.isPreviewInitiated) return;
-    const _requiredField = state.widgetRequiredFieldSchemaMap.map(([d]) => d);
-    if (_requiredField.every((d) => widgetGenerateState.widgetValidMap[d])) {
-        state.isPreviewInitiated = true;
-    }
-}, { deep: true });
-
```
Comment (piggggggggy) [inline apps/web/src/common/modules/widgets/_components/WidgetFormOverlayStep2WidgetForm.vue:195]: Oh Thanks~

**Review by piggggggggy — APPROVED** [review]
LGTM~

**Review by sulmoJ — APPROVED** [review]
goooood

**Pull request merged**